elit0451/SI-FinalProject
# Rating service - Issues #36 #37

## Checklist
- [x] Subscribe to "**event.feedback**" queue while listening to either an "Add" or "Get" command 
- [x] Connect to a SQL Database

### PR Description
Created a new .Net Core console application.
Created a DBSchema for the service, which is described in "DBSCHEMA" file located in the root.
Added MessageGateway.cd to communicating with RabbitMQ.
Added Rating model which corresponds to the Rating entity in the DB.
Added functionality subscribe to "event.feedback" queue.
Implemented a Command pattern in CommandRouter to distinguish b/n "Add" or "Get" command.
Added Dockerfile to build an image from Rating service.

### How to test
I have created a docker image of the rating service and it is pushed to docker hub

Start by executing the following docker commands:
- `docker network create testnet`
- `docker run --rm --name mysql -d -v ${PWD}/db_data:/var/lib/mysql -p 3306:3306 --network=testnet -e MYSQL_ROOT_PASSWORD=soft2019Backend mysql`
**Make sure to execute the DBSCHEMA file against the database to have the proper data.**

- `docker run --rm --name rabbitmq -d --network=testnet -p 15672:15672 -p 5672:5672 rabbitmq:management`
- `docker run --rm --name ratingservice --network=testnet davi7816/testratingservice`
(before succeeding to start the service, you should give some time to rabbitmq to start)

After all of these are started, you can test the connector functionality by heading to the rabbit MQ web interface at http://localhost:15672/. In there head over to queues and create a new queue with a name of your choice, _ex: test_ by keeping add the defaults. You should also see a event.feedback queue, click it.
When the page loads you should see a publish area where you can insert any of the following commands:
> Adding a rating  

`{"EventId":5,"Feedback":3,"command":"add"}`
This will create a new event rating. You can verify that the rating has been persisted to the DB by executing the following query in MySQL Workbench: `SELECT * FROM RatingService.Rating;`

> Getting a rating

In the same **events.feedback** queue copy the following payload:
`{"EventId":5,"command":"get"}`
You also need to add 2 Properties: `correlation_id ` with value `guid` and `reply_to ` with value `test` or the queue name that you created previously.
You can check the response by heading over to that queue and pressing "Get message(s)"

Issue reference

close #36 
close #37
